include/deal.II/matrix_free/tools.h

```diff
@@ -157,6 +157,179 @@ namespace MatrixFreeTools
     const unsigned int first_selected_component = 0);
 
 
+
+  /**
+   * A wrapper around MatrixFree to help users to deal with DoFHandlers,
+   * where there are cells without degrees of freedom, i.e., with
+   * FENothing assigned to them. In the following we call such cells
+   * invalid. All other cells are valid. In contrast to MatrixFree,
+   * this class skips invalid cells and faces between valid and
+   * invalid cells are treated as boundary faces.
```

Comment:
Maybe mention how to figure out if a face is an actual boundary face or a valid-invalid boundary face inside the user-supplied boundary operation?

Reply:
I would use this function: https://github.com/dealii/dealii/pull/14122 :)